### DISCUSSION TITLE
is there a way to change case of field names

### DISCUSSION THREAD

**Discussion by tshepang** [vectordotdev/vector#11643] [Q&A] [answered]
Am looking to translate a logstash filter which has a kv filter using [`transform_key`](https://www.elastic.co/guide/en/logstash/master/plugins-filters-kv.html#plugins-filters-kv-transform_key).

**Comment by jszwedko** [comment, marked as answer]
Not currently, but this is a use-case that would be supported by iteration in VRL (which is being worked on in #10985 ). We may still need some additional VRL functions though, like `lowercase`. Would you want to open an issue with the ones you need?

As a workaround, the `lua` transform can be used.

**Reply by tshepang** [reply]
not sure a separate issue is needed... looks to me like iteration would support such automatically since all keys are strings (I assume)

**Reply by jszwedko** [reply]
Thanks for confirming. Even with iteration we would still need string manipulation functions like `lowercase` though. I'm curious which functions you'd benefit from.

**Reply by tshepang** [reply]
it's just `lowercase` as far as I've seen
